google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/StreamWriter.java

```diff
@@ -265,6 +280,33 @@ public Boolean expired() {
     return outstandingAppend.appendResult;
   }
 
+  /**
+   * This is the general flush method for asynchronise append operation. When you have outstanding
+   * append requests, calling flush will make sure all outstanding append requests completed and
+   * successful. Otherwise there will be an exception thrown.
+   *
+   * @throws Exception
+   */
+  public void flushAll(long timeoutMillis) throws Exception {
+    appendAndRefreshAppendLock.lock();
+    try {
+      writeAllOutstanding();
+      synchronized (messagesWaiter) {
+        messagesWaiter.waitComplete(timeoutMillis);
```

Comment:
should we return deadline exceed error if timeout is reached?

Reply:
It seems we are throwing InterruptedException on timeouts, see awaitTermination function.